services/auth/auth.go

```diff
@@ -0,0 +1,161 @@
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package auth provides authentication/authorization functionality.
+package auth
+
+import (
+	"context"
+	"net"
+	"os"
+	"time"
+
+	api "github.com/percona-platform/saas/gen/auth/external"
+	"github.com/percona/pmm/utils/tlsconfig"
+	"github.com/percona/pmm/version"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+const (
+	defaultHost = "check.percona.com:443"
+	envHost     = "PERCONA_TEST_AUTH_HOST"
+	timeout     = 5 * time.Second
+)
+
+// Service is responsible for interactions with Percona authentication service.
+type Service struct {
+	db   *reform.DB
+	host string
+	l    *logrus.Entry
+}
+
+// New returns auth Service.
+func New(db *reform.DB) *Service {
+	l := logrus.WithField("component", "auth")
+
+	s := Service{
+		host: defaultHost,
+		db:   db,
+		l:    l,
+	}
+
+	if h := os.Getenv(envHost); h != "" {
+		l.Warnf("Host changed to %s.", h)
+		s.host = h
+	}
+
+	return &s
+}
+
+// SignUp creates new user with given email and password.
+func (s *Service) SignUp(ctx context.Context, email, password string) error {
+	cc, err := s.getConnection(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed establish connection with Percona")
+	}
+	defer cc.Close() //nolint:errcheck
+
+	_, err = api.NewAuthAPIClient(cc).SignUp(ctx, &api.SignUpRequest{Email: email, Password: password})
+	if err != nil {
+		return errors.Wrap(err, "failed to sign up user")
+	}
+
+	return nil
+}
+
+// SignIn creates new session for user with given email and password.
+func (s *Service) SignIn(ctx context.Context, email, password string) error {
+	cc, err := s.getConnection(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed establish connection with Percona")
+	}
+	defer cc.Close() //nolint:errcheck
+
+	resp, err := api.NewAuthAPIClient(cc).SignIn(ctx, &api.SignInRequest{Email: email, Password: password})
+	if err != nil {
+		return errors.Wrap(err, "failed to sign in user")
+	}
+
+	err = s.db.InTransaction(func(tx *reform.TX) error {
+		_, err := models.UpdateSettings(tx.Querier, &models.ChangeSettingsParams{SessionID: resp.SessionId})
+		return err
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to save session id")
+	}
+
+	return nil
+}
+
+// RefreshSession resets session timeout.
+func (s *Service) RefreshSession(ctx context.Context) error {
+	settings, err := models.GetSettings(s.db)
+	if err != nil {
+		return err
+	}
+
+	if settings.SessionID == "" {
+		return errors.New("no active sessions")
+	}
+
+	cc, err := s.getConnection(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed establish connection with Percona")
+	}
+	defer cc.Close() //nolint:errcheck
+
+	md := metadata.New(map[string]string{"authorization": "PP-v1beta1 " + settings.SessionID})
```

Comment:
Let's move "PP-v1beta1" to the named constant

Reply:
Done